drivers/spi/spi_mcux_lpspi.c

```diff
@@ -286,27 +286,27 @@ static struct spi_mcux_data spi_mcux_data_0 = {
 	SPI_CONTEXT_INIT_SYNC(spi_mcux_data_0, ctx),
 };
 
-DEVICE_AND_API_INIT(spi_mcux_0, SPI_0_LABEL, &spi_mcux_init,
+DEVICE_AND_API_INIT(spi_mcux_0, DT_NXP_IMX_LPSPI_SPI_0_LABEL, &spi_mcux_init,
 		    &spi_mcux_data_0, &spi_mcux_config_0,
 		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
 		    &spi_mcux_driver_api);
 
 static void spi_mcux_config_func_0(struct device *dev)
 {
-	IRQ_CONNECT(SPI_0_IRQ, SPI_0_IRQ_PRIORITY,
+	IRQ_CONNECT(DT_NXP_IMX_LPSPI_SPI_0_IRQ, DT_NXP_IMX_LPSPI_SPI_0_IRQ_PRIORITY,
 		    spi_mcux_isr, DEVICE_GET(spi_mcux_0), 0);
 
-	irq_enable(SPI_0_IRQ);
+	irq_enable(DT_NXP_IMX_LPSPI_SPI_0_IRQ);
 }
 #endif /* SPI_0 */
 
 #ifdef CONFIG_SPI_1
```

Comment:
It looks like we are using the `DT_<COMPAT>_<INSTANCE>_<PROP>` pattern. In this case we are not able any more to enable instances via Kconfig. This should be changed to `DT_NXP_IMX_LPSPI_SPI_1`. Also updates to the code controlling pin muxing seems to be missing.

Reply:
Oddly, this isn't the `DT_<COMPAT>_<INSTANCE>_<PROP>`.  Its using a define based on the label, which matches what SPI_0_IRQ was defined to.

Reply:
Ah, OK. I was not aware that the PR to generate defines based on label got merged.

Reply:
Its hard to remember, just trying to move this forward for LTS.  Once pinmux gets solved that will make it much easier to move to `DT_<COMPAT>_<INSTANCE>_<PROP>` for most drivers.

Reply:
No, the define isn't coming from the label, it's coming from the alias in the board dts.
https://github.com/zephyrproject-rtos/zephyr/blob/d2886ab8bc49f9bbc8bc7461e36698ecea60e7a4/boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts#L15-L28